dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/UseCallingThreadExecutor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.dialogue.serde;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.random.SafeThreadLocalRandom;
+
+/**
+ * Package private API for enabling
+ * {@link com.palantir.dialogue.blocking.CallingThreadExecutor} optimization.
+ */
+final class UseCallingThreadExecutor {
+
+    private static final AtomicDouble PROBABILITY = new AtomicDouble(0.0);
```

Comment:
nit: volatile since we're not doing any sort of CAS where we need the padding. float type should be sufficient, I don't imagine we expect a ton of decimal points.

Reply:
done